### PR TITLE
Sent orders

### PR DESCRIPTION
**Pull request by ennaque**

### CODE REVIEW

File: src/app/my-orders/components/inbox/inbox.component.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    constructor(
+        private readonly receivedOrdersApi: ReceivedOrdersApiService,
+        private readonly changeDetector: ChangeDetectorRef
+    ) {
+    }
+
+    public ngOnInit(): void {
+        this.currentFilter$.pipe(
+            takeUntil(this.destroy$),
```
Comment (vitslepukhin) [inline src/app/my-orders/components/inbox/inbox.component.ts:29]: The operator takeUntil must be last in sequence in sake of unsubscription

File: src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Input()
+    public set order(order: SentOrder) {
+        this._order = order;
+        if (!order) {
+            return;
+        }
+        this.servicesCache.getById(order.service).pipe(
+            switchMap(service => this.masterCache.getById(service.professional)),
+            tap(master => this.master = master),
```
Comment (vitslepukhin) [inline src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.ts:42]: This is very strange using of tap operators. Why didn't you put it into subscribe?

File: src/app/my-orders/my-orders-page.component.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<app-header></app-header>
+<ion-content>
+    <ion-grid class="ion-no-padding">
+        <ion-row class="ion-justify-content-center">
+            <ion-col sizeLg="7" sizeMd="8" sizeSm="8" sizeXl="5" sizeXs="12">
```
Comment (vitslepukhin) [inline src/app/my-orders/my-orders-page.component.html:5]: Is directive appMainColumn fits here? (for set column sizes)

File: src/polyfills.ts [hunk 4]
```diff
@@ -69,5 +69,7 @@ import 'zone.js/dist/zone';  // Included with Angular CLI.
  * APPLICATION IMPORTS
  */
 
+import 'core-js/es7/reflect';
```
Comment (enkryptor) [inline src/polyfills.ts:72]: let's add comment "is needed for class-transformer"

File: src/app/my-orders/services/received-orders-api.service.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+export class ReceivedOrdersApiService extends AbstractApiService<ReceivedOrder> {
+    constructor(client: ApiClientService) {
+        super(client);
+    }
+
+    protected getUrl(): string {
+        return environment.backend.received_orders;
+    }
+
+    // @ts-ignore
```
Comment (enkryptor) [inline src/app/my-orders/services/received-orders-api.service.ts:20]: We can remove ` @ts-ignore` here. It's what the third signature was needed for.

File: src/app/core/abstract/abstract-readonly-api.service.ts [hunk 6]
```diff
@@ -37,9 +37,6 @@ export abstract class AbstractReadonlyApiService<T> implements ReadonlyApiServic
 
     protected abstract getUrl(): string;
 
-    protected transform(data: T): T;
-    protected transform(data: T[]): T[];
-    protected transform(data: T | T[]): T | T[] {
```
Comment (enkryptor) [inline src/app/core/abstract/abstract-readonly-api.service.ts:42]: Please leave the third signature intact (can remove the implementation though).

File: src/app/my-orders/my-orders-page.component.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import {Component, OnInit} from '@angular/core';
+import {MasterManagerService} from '@app/core/services';
+import {MasterReadonlyApiCacheService} from '@app/my-orders/services/master-readonly-api-cache.service';
+import {ServicesApiCache} from './services';
+
+@Component({
+    selector: 'app-my-orders-page',
+    templateUrl: './my-orders-page.component.html',
+    styleUrls: ['./my-orders-page.component.scss'],
+    providers: [ServicesApiCache, MasterReadonlyApiCacheService]
```
Comment (vitslepukhin) [inline src/app/my-orders/my-orders-page.component.ts:10]: Do you really need the ServicesApiCache here? It isn't used in component.

File: src/app/my-orders/components/outbox/outbox.component.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        this.sub = this.currentFilter$.pipe(
+            switchMap(params => this.sentOrdersApi.get(params))
+        ).subscribe(orders => {
+            this.orders = orders.results;
+            this.changeDetector.markForCheck();
+        });
+    }
+
+    public ngOnDestroy(): void {
+        this.sub.unsubscribe();
```
Comment (enkryptor) [inline src/app/my-orders/components/outbox/outbox.component.ts:36]: Let's use `takeUntil()` instead

File: src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.scss [hunk 9]
```diff
@@ -0,0 +1,4 @@
+.company {
+  color: #86888f;
```
Comment (enkryptor) [inline src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.scss:2]: We should support theming.

Comment (ennaque) [reply src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.scss:2]: this unit will be grey anyway

File: src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.ts (hunk 2 above)
Comment (enkryptor) [inline src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.ts:42]: Let's move this logic inside the `subscribe()`

File: src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.ts [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+            tap(() => this.changeDetector.markForCheck())
+        ).subscribe();
```
Comment (enkryptor) [inline src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.ts:44]: Maybe use `async` pipe instead?

File: src/app/my-orders/my-orders-page.component.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+})
+export class MyOrdersPageComponent implements OnInit {
+    public isInbox: boolean = false;
+    public isMaster: boolean;
+
+    constructor(private readonly masterManager: MasterManagerService) {
+    }
+
+    public ngOnInit(): void {
+        this.masterManager.isMaster().subscribe(res => this.isMaster = res);
```
Comment (enkryptor) [inline src/app/my-orders/my-orders-page.component.ts:20]: Leaked subscription. We should unsubscribe (by using `async` pipe, for instance)

File: src/app/order/services/sent-orders-api.service.ts [hunk 12]
```diff
@@ -16,4 +17,9 @@ export class OrdersApiService extends AbstractApiService<OrderModel> {
     protected getUrl(): string {
         return environment.backend.sent_orders;
     }
+
+    // @ts-ignore
```
Comment (enkryptor) [inline src/app/order/services/sent-orders-api.service.ts:21]: Let's not use `@ts-ignore`

File: src/app/my-orders/my-orders-page.component.ts (hunk 11 above)
Comment (ennaque) [reply src/app/my-orders/my-orders-page.component.ts:20]: its one time observable

**Review by enkryptor — CHANGES_REQUESTED** [review]
1. Don't use `@ts-ignore`, fix the typing instead
2. Use takeUntil() in place of manual unsubscribing
3. Don't use `tap`. Use async pipe when appropriate

File: src/app/my-orders/my-orders-page.component.ts (hunk 7 above)
Comment (ennaque) [reply src/app/my-orders/my-orders-page.component.ts:10]: it is

File: src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.scss (hunk 9 above)
Comment (enkryptor) [reply src/app/my-orders/components/sent-order-list-item/sent-order-list-item.component.scss:2]: We already have the pre-defined color for grey

File: src/app/my-orders/components/order-status/order-status.component.html [hunk 13]
```diff
@@ -0,0 +1,10 @@
+<ng-container [ngSwitch]="status">
+    <ion-button *ngSwitchCase="'new'" [disabled]="true"
+                class="opacity-full" color="warning">{{ 'order-status.pending' | translate }}</ion-button>
+    <ion-button *ngSwitchCase="'confirmed'" [disabled]="true"
+                class="opacity-full" color="success">{{ 'order-status.accepted' | translate }}</ion-button>
+    <ion-button *ngSwitchCase="'archived'" [disabled]="true"
+                class="opacity-full" color="dark">{{ 'order-status.archived' | translate }}</ion-button>
+    <ion-button *ngSwitchCase="'completed'" [disabled]="true"
+                class="opacity-full" color="medium">{{'order-status.completed'|translate}}</ion-button>
```
Comment (enkryptor) [inline src/app/my-orders/components/order-status/order-status.component.html:9]:
```suggestion
                class="opacity-full" color="medium">{{ 'order-status.completed' | translate }}</ion-button>
```

**Review by enkryptor — APPROVED** [review]

File: src/app/my-orders/components/inbox/inbox.component.ts (hunk 1 above)
Comment (ennaque) [reply src/app/my-orders/components/inbox/inbox.component.ts:29]: one time observable is called next in sequence anyway

**Pull request merged**